### PR TITLE
feat: Add send max when create multisig transaction

### PR DESCRIPTION
**Pull request by yanguoyu**
https://user-images.githubusercontent.com/12881040/161878584-7dad5ade-f6ea-42a1-886b-e1a6b7162dcf.mov

### CODE REVIEW

File: packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts [hunk 1]
```diff
@@ -118,28 +140,76 @@ export const useSendInfo = ({
   const dispatch = useDispatch()
   const [errorMessage, setErrorMessage] = useState('')
   useEffect(() => {
-    if (outputErrors.some(v => v.addrError || v.amountError)) {
-      return
-    }
     clearTimeout(generateTxTimer)
     setErrorMessage('')
-    const validSendInfoList = sendInfoList.filter(v => v.address && v.amount)
+    const validSendInfoList = sendInfoList.filter(
```
Comment (Keith-CY) [inline packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts:145]: If addresses and amounts will be verified in `generateMultisigTxWith`, why should we filter invalid addresses and amounts?

File: packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts [hunk 2]
```diff
@@ -118,28 +140,76 @@ export const useSendInfo = ({
...
         })
       }
     }, 300)
-  }, [sendInfoList, setErrorMessage, multisigConfig, dispatch, t, outputErrors])
+  }, [sendInfoList, setErrorMessage, multisigConfig, dispatch, t, isMainnet, outputErrors])
+  const [isSendMax, setIsSendMax] = useState(false)
+  const onSendMaxClick = useCallback(
+    e => {
+      const {
+        dataset: { isOn = 'false' },
```
Comment (Keith-CY) [inline packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts:169]: What does `isOn` mean?

File: packages/neuron-wallet/src/services/cells.ts [hunk 3]
```diff
@@ -620,16 +621,21 @@ export default class CellsService {
         output.status IN (:...statuses) AND
         hasData = false AND
         typeHash is null AND
-        output.lockArgs in (
-          SELECT publicKeyInBlake160
-          FROM hd_public_key_info
-          WHERE walletId = :walletId
-        ) AND
+        ${
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/cells.ts:624]: Tests should be updated

File: packages/neuron-wallet/src/services/transaction-sender.ts [hunk 4]
```diff
@@ -440,6 +440,26 @@ export default class TransactionSender {
     return tx
   }
 
+  public generateMultisigSendAllTx = async (
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/transaction-sender.ts:443]: Tests are missing

File: packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts (hunk 2 above)
Comment (yanguoyu) [reply packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts:169]: It equal with the `isSendMax`

File: packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts (hunk 1 above)
Comment (yanguoyu) [reply packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts:145]: It avoid the valid outputs's length is 0, but it also has transaction fee.

**Comment by Keith-CY** [comment]
image
image
I triggered an error when synchronization is not fully done.

File: packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts (hunk 2 above)
Comment (Keith-CY) [reply packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts:169]: What does `setIsSendMax(isOn === 'false')` mean if `isSendMax` equals to `isSendMax`

Comment (Keith-CY) [reply packages/neuron-ui/src/components/SendFromMultisigDialog/hooks.ts:169]: Ah, I understand it, when `isOn` is `false`, by clicking the button, `isSendMax` is set to `true`

**Comment by yanguoyu** [comment]
> image image I triggered an error when synchronization is not fully done.

Fixed by judging whether `res` is undefined

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**